src/hooks/useCanvas/const.ts

```diff
@@ -12,26 +12,14 @@ export const setCanvasHeight = () => {
   return document.body.clientHeight - 150
 }
 
-export const handleSetTool = (data: handleSetToolParams) => {
-  const { canvasRef, roomId, socket, tool } = data
-
-  switch (tool) {
-    case DrawTools.PEN:
-      new Pen(canvasRef, socket, roomId)
-      break
-    case DrawTools.SQUARE:
-      new Square(canvasRef, socket, roomId)
-      break
-    case DrawTools.CIRCLE:
-      new Circle(canvasRef, socket, roomId)
-      break
-    case DrawTools.ERASER:
-      new Eraser(canvasRef, socket, roomId)
-      break
-    case DrawTools.LINE:
-      new Line(canvasRef, socket, roomId)
-      break
-    default:
-      new Pen(canvasRef, socket, roomId)
+export const handleSetTool = ({ canvasRef, roomId, socket, tool }: handleSetToolParams) => {
+  const draw = {
+    [DrawTools.PEN]: () => new Pen(canvasRef, socket, roomId),
+    [DrawTools.SQUARE]: () => new Square(canvasRef, socket, roomId),
+    [DrawTools.CIRCLE]: () => new Circle(canvasRef, socket, roomId),
+    [DrawTools.ERASER]: () => new Eraser(canvasRef, socket, roomId),
+    [DrawTools.LINE]: () => new Line(canvasRef, socket, roomId)
```

Comment:
leave here only class and use `new` after tool will selected from object. After this change you can move object out of function